### PR TITLE
feature/trend_vision_micro_one_oat_add_fields

### PR DESCRIPTION
**Pull request by clement-burtscher-sekoia**
## Description

Fixes https://github.com/SekoiaLab/integration/issues/1658

### Added

- Parse new dynamic fields (ECS):
    - destination.as.organization.name
    - destination.domain
    - destination.geo.country_name
    - destination.ip
    - destination.mac
    - destination.port
    - event.severity
    - file.name
    - network.application
    - observer.name
    - rule.id
    - source.as.organization.name
    - source.geo.city_name
    - source.geo.country_name
    - source.ip
    - source.mac
    - source.port
- Parse new custom fields:
    - trendmicro.visionone.oat.attackphase.source
    - trendmicro.visionone.oat.attackphase.target
    - trendmicro.visionone.oat.count
    - trendmicro.visionone.oat.malName
    - trendmicro.visionone.oat.malType
    - trendmicro.visionone.oat.riskLevel (extended fallback mapping)
    - trendmicro.visionone.oat.threatType

### Changed

- Complete parsing of existing dynamic fields (ECS):
    - event.provider
    - host.name
    - observer.product
    - process.executable
    - rule.name
    - threat.tactic.id
    - trendmicro.visionone.oat.riskLevel
    - url.original
    - user.domain
    - user.name

## Type of Change

- [ ] New intake format
- [X] Update to existing intake format
- [ ] Bug fix
- [ ] Documentation update
- [ ] Other (please describe):

## Affected Intake Formats

- Vendor/product:

### 🔒 CRITICAL - Security Requirements (MUST BE VERIFIED FIRST)

- [ ] **No real email addresses** in test files (use: user@example.com, test@example.org)
- [ ] **No real passwords or API keys** (use: "P@ssw0rd!", "xxxxxxxxxxxx", "FAKE_TOKEN_123")
- [ ] **No real IP addresses** (use TEST-NET ranges: 192.0.2.x, 198.51.100.x, 203.0.113.x)
- [ ] **No real phone numbers** (use: +1-555-0100 or similar test numbers)
- [ ] **No real names or PII** (use: "John Doe", "Jane Smith", "User123")
- [ ] **No real company/organization names** in test data (unless public vendors)
- [ ] **No real usernames, hostnames, or domain names** (use: user1, host.example.com, test.corp)
- [ ] All test data has been anonymized and verified

### Required for All PRs

- [ ] Code has been linted
    - [ ] with the linter for manifest, smart-descriptions, and test files (`uv run python linter.py check --changes` in utils/)
    - [ ] with Prettier for the parser code files (`npx prettier --write <module>/<format>/ingest/parser.yml`)
- [ ] All CI/CD checks pass
- [ ] Changes follow the contribution guidelines in `CONTRIBUTING.md`

### Required for New/Updated Intake Formats

- [ ] Clear descriptions provided for modules and formats
- [ ] Logo files included for new modules/formats
- [ ] Parser test coverage is at least 75%
- [ ] At least one smart-description is provided
- [ ] README.md updated (if applicable)
- [ ] Taxonomy mappings are correct and documented
- [ ] Test cases cover edge cases and error handling

### Testing

- [ ] Local tests pass (`uv run pytest` in utils/)
- [ ] Sample data is representative and **completely anonymized**
- [ ] **No sensitive information** (PII, credentials, real IPs, real emails) in test files
- [ ] Test data uses example.com/example.org domains and TEST-NET IP ranges
- [ ] Parser handles malformed data gracefully

### Documentation

- [ ] Code changes are documented
- [ ] Smart-descriptions are clear and informative
- [ ] Field mappings are explained

## Related Issues

Closes #

## Summary by Sourcery

Extend Trend Micro Vision One OAT intake mappings with additional ECS and custom fields and improve existing field population for richer events.

New Features:
- Map additional ECS networking, host, process, rule, and severity fields for Vision One OAT events, including destination/source attributes, network.application, observer details, and file.name.
- Introduce new custom Vision One OAT fields for attack phases, event count, malware name/type, threat type, and enhanced risk level handling.

Enhancements:
- Refine existing mappings for provider, host, observer.product, process.executable, rule.name, threat.tactic.id, url.original, user domain/name, and rule IDs to use more complete fallback sources.
- Factor out custom-field handling into a dedicated parser stage to clarify the pipeline structure.
- Update Vision One OAT changelog with the newly added and revised field mappings.

Tests:
- Add and update observed attack technique test fixtures to cover the new and updated field mappings.

### CODE REVIEW

File: Trend Micro/trend-micro-vision-one-oat/ingest/parser.yml [hunk 1]
```diff
@@ -28,80 +30,85 @@ stages:
...
-          host.name: "{{parsed_event.message.endpoint.endpointName}}"
-          host.ip: "{{parsed_event.message.endpoint.ips}}"
-
+          action.properties.ScriptBlockText: "{{parsed_event.message.detail.objectRawDataStr}}"
           agent.id: "{{parsed_event.message.endpoint.agentGuid}}"
-          event.start: "{{parsed_event.message.detail.firstSeen | to_rfc3339}}"
+          cloud.service.name: "{{parsed_event.message.cloudAppName}}"
+          destination.as.organization.name: "{{parsed_event.message.detail.dstIsp}}"
+          destination.domain: "{{parsed_event.message.detail.peerHost or parsed_event.message.detail.cccaDestination or parsed_event.message.detail.denyListHost}}"
+          destination.geo.country_name: "{{parsed_event.message.detail.dstCountry or parsed_event.message.detail.dstIsp}}"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/ingest/parser.yml]: Fixed with commit [Apply Copilot review requested changes](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/cf1ed19299cfc9332232979f538f33b6d2768bf6)

File: Trend Micro/trend-micro-vision-one-oat/ingest/parser.yml [hunk 2]
```diff
@@ -28,80 +30,85 @@ stages:
...
-
-          organization.id: "{{parsed_event.message.orgId}}"
-
+          rule.id: "{{parsed_event.message.detail.ruleId}}"
+          rule.name: "{{parsed_event.message.ruleName or parsed_event.message.detail.ruleName or (parsed_event.message.filters | map(attribute='name') | list if parsed_event.message.filters and parsed_event.message.filters | length > 0 else none)}}"
           rule.ruleset: "{{parsed_event.message.policyName}}"
-          rule.name: "{{parsed_event.message.ruleName}}"
+          source.as.organization.name: "{{parsed_event.message.detail.srcIsp}}"
+          source.geo.city_name: "{{parsed_event.message.detail.srcCity}}"
+          source.geo.country_name: "{{parsed_event.message.detail.srcCountry or parsed_event.message.detail.srcCity or parsed_event.message.detail.srcIsp}}"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/ingest/parser.yml]: Fixed with commit [Apply Copilot review requested changes](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/cf1ed19299cfc9332232979f538f33b6d2768bf6)

File: Trend Micro/trend-micro-vision-one-oat/ingest/parser.yml [hunk 3]
```diff
@@ -28,80 +30,85 @@ stages:
...
           host.os.name: "{{parsed_event.message.detail.osName}}"
           host.os.version: "{{parsed_event.message.detail.osVer}}"
-          host.os.full: "{{parsed_event.message.detail.osDescription}}"
-
+          network.application: "{{parsed_event.message.detail.app}}"
+          observer.name: "{{parsed_event.message.detail.dvchost}}"
+          observer.product: "{{parsed_event.message.get('detail', {}).get('dvchost', 'Vision One')}}"
+          organization.id: "{{parsed_event.message.orgId}}"
+          process.command_line: "{{parsed_event.message.detail.objectCmd}}"
+          process.executable: "{{parsed_event.message.detail.ObjectFilePath or parsed_event.message.detail.filePathName}}"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/ingest/parser.yml]: Fixed with commit [Apply Copilot review requested changes](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/cf1ed19299cfc9332232979f538f33b6d2768bf6)

File: Trend Micro/trend-micro-vision-one-oat/ingest/parser.yml [hunk 4]
```diff
@@ -120,12 +127,28 @@ stages:
 
         filter: "{{parsed_event.message.filters | length > 0 }}"
 
+  set_custom_fields:
+    actions:
+      - set:
+          trendmicro.visionone.oat.attackphase.source: "{{parsed_event.message.detail.sAttackPhase}}"
+          trendmicro.visionone.oat.attackphase.target: "{{parsed_event.message.detail.pAttackPhase}}"
+          trendmicro.visionone.oat.count: "{{parsed_event.message.detail.cnt}}"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/ingest/parser.yml]: Fixed with commit [Apply Copilot review requested changes](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/cf1ed19299cfc9332232979f538f33b6d2768bf6)

File: Trend Micro/trend-micro-vision-one-oat/ingest/parser.yml [hunk 5]
```diff
@@ -28,80 +30,85 @@ stages:
...
-          process.executable: "{{parsed_event.message.detail.ObjectFilePath}}"
-          process.hash.md5: "{{parsed_event.message.detail.ObjectFileHashMd5}}"
-          process.hash.sha1: "{{parsed_event.message.detail.ObjectFileHashSha1}}"
-          process.hash.sha256: "{{parsed_event.message.detail.ObjectFileHashSha256}}"
-
-          url.original: "{{ parsed_event.message.request }}"
-
-          organization.id: "{{parsed_event.message.orgId}}"
-
+          rule.id: "{{parsed_event.message.detail.ruleId}}"
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/ingest/parser.yml]: Fixed with commit [Apply Copilot review requested changes](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/cf1ed19299cfc9332232979f538f33b6d2768bf6)

File: Trend Micro/trend-micro-vision-one-oat/CHANGELOG.md [hunk 6]
```diff
@@ -6,3 +6,48 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+## [Unreleased] - 2026-06-11
+
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/CHANGELOG.md:11]: Cannot version intake-formats changelogs so keep it like this

File: Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_1.json [hunk 7]
```diff
@@ -10,6 +10,7 @@
       ],
       "dataset": "endpointActivityData",
       "end": "2022-04-12T23:43:15Z",
+      "provider": "751",
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_1.json:13]: Fixed with commit [Anonymize tests](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/2f8570669f14a4f928270ccfb5711ea52695fd1f)

File: Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_2.json [hunk 8]
```diff
@@ -10,6 +10,7 @@
       ],
       "dataset": "endpointActivityData",
       "end": "2024-11-26T16:45:02.571000Z",
+      "provider": "751",
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_2.json:13]: Fixed with commit [Anonymize tests](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/2f8570669f14a4f928270ccfb5711ea52695fd1f)

File: Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_7.json [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+  },
+  "expected": {
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_7.json:5]: Fixed with commit [Anonymize tests](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/2f8570669f14a4f928270ccfb5711ea52695fd1f)

File: Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_9.json [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+  },
+  "expected": {
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_9.json:5]: Fixed with commit [Anonymize tests](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/2f8570669f14a4f928270ccfb5711ea52695fd1f)

File: Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_3.json [hunk 11]
```diff
@@ -10,6 +10,7 @@
       ],
       "dataset": "endpointActivityData",
       "end": "2024-11-26T16:45:03.446000Z",
+      "provider": "751",
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_3.json:13]: Fixed with commit [Anonymize tests](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/2f8570669f14a4f928270ccfb5711ea52695fd1f)

File: Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_8.json [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+  },
+  "expected": {
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_8.json:5]: Fixed with commit [Anonymize tests](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/2f8570669f14a4f928270ccfb5711ea52695fd1f)

File: Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_10.json [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+  },
+  "expected": {
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_10.json:5]: Fixed with commit [Anonymize tests](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/2f8570669f14a4f928270ccfb5711ea52695fd1f)

File: Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_11.json [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+  },
+  "expected": {
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_11.json:5]: Fixed with commit [Anonymize tests](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/2f8570669f14a4f928270ccfb5711ea52695fd1f)

File: Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_12.json [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+  },
+  "expected": {
```
[the remark this answers is not in the document]
Comment (clement-burtscher-sekoia) [reply Trend Micro/trend-micro-vision-one-oat/tests/test_observed_attack_technique_12.json:5]: Fixed with commit [Anonymize tests](https://github.com/SEKOIA-IO/intake-formats/pull/2609/changes/2f8570669f14a4f928270ccfb5711ea52695fd1f)

**Review by mchupeau-sk — APPROVED** [review]
LGFM
